bridge/server/services/data-service.ts

```diff
@@ -53,87 +60,218 @@ export class DataService {
+    return Project.fromJSON(response.data);
+  }
+
   public async getProject(
     projectName: string,
     includeRemediation: boolean,
     includeApproval: boolean
   ): Promise<Project> {
-    const response = await this.apiService.getProject(projectName);
-    const project = Project.fromJSON(response.data);
-    let remediations: Remediation[] = [];
+    const project = await this.getPlainProject(projectName);
+    let openApprovals: Approval[] = [];
+    let openRemediations: Remediation[] = [];
+    const allServices = Stage.getAllServices(project.stages);
+    const latestDeployments = await this.getDeploymentFinishedForServices(projectName, allServices, ResultTypes.PASSED);
+    const latestEvaluations = await this.getEvaluationResultsForServices(projectName, allServices);
+    // for sequence adjustment: const latestSequences = await this.getSequencesForServices(projectName, allServices);
+    const cachedSequences: { [keptnContext: string]: Sequence | undefined } = {};
 
     if (includeRemediation) {
-      remediations = await this.getOpenRemediations(projectName, true, true);
+      openRemediations = await this.getOpenRemediations(projectName, true, true);
+    }
+
+    if (includeApproval) {
+      openApprovals = await this.getApprovals(projectName, true);
     }
-    const lastSequences: { [key: string]: Sequence } = {};
+
     for (const stage of project.stages) {
+      const stageRemediations = openRemediations.filter((seq) => seq.stages.some((s) => s.name === stage.stageName));
+      const stageApprovals = openApprovals.filter((approval) => approval.trace.data.stage === stage.stageName);
+      const stageEvaluations = latestEvaluations.filter((t) => t.data.stage === stage.stageName);
+      const stageDeployments = latestDeployments.filter((t) => t.data.stage === stage.stageName);
+      // for sequence adjustment: const stageSequences = latestSequences.filter((seq) => seq.stages.some((st) => st.name === stage.stageName));
+      const stageInformation: StageOpenInformation = {
+        openApprovals: stageApprovals,
+        openRemediations: stageRemediations,
+        evaluations: stageEvaluations,
+      };
+
       for (const service of stage.services) {
-        const keptnContext = service.getLatestSequence();
-        if (keptnContext) {
-          try {
-            const latestSequence = await this.fetchServiceDetails(
-              service,
-              stage.stageName,
-              keptnContext,
-              projectName,
-              includeApproval,
-              remediations,
-              lastSequences[service.serviceName]
-            );
-            if (latestSequence) {
-              lastSequences[service.serviceName] = latestSequence;
-            }
-          } catch (error) {
-            console.error(error);
+        const latestEvent = service.getLatestEvent();
+        if (latestEvent) {
+          const latestSequence = await this.fetchServiceDetails(
```

Comment:
Maybe naming, but by calling just this it's a bit hard to say, that this call mutates the service object in the project